maestrowf/datastructures/schemas.json

```diff
@@ -0,0 +1,116 @@
+    },
+    "additionalProperties": false,
+    "required": [
+      "name",
+      "description"
+    ]
+  },
+  "PARAM": {
+    "type": "object",
+    "properties": {
+      "values": {
+        "type": "array"
+      },
+      "label": {"type": "string", "minLength": 1}
+    },
+    "additionalProperties": false,
+    "required": [
+      "values",
+      "label"
+    ]
+  },
+  "STUDY_STEP": {
+    "type": "object",
+    "properties": {
+      "name": {"type": "string", "minLength": 1},
+      "description": {"type": "string", "minLength": 1},
+      "run": {
+        "type": "object",
+        "properties": {
+          "cmd": {"type": "string", "minLength": 1},
+          "depends": {"type": "array", "uniqueItems": true},
+          "pre": {"type": "string", "minLength": 1},
+          "post": {"type": "string", "minLength": 1},
+          "restart": {"type": "string", "minLength": 1},
+          "nodes": {"type": "number"},
+          "procs": {"type": "number"},
+          "gpus": {"type": "number"},
+          "cores per task": {"type": "number"},
+          "walltime": {"type": "string", "minLength": 1},
+          "reservation": {"type": "string", "minLength": 1},
+          "required": [
+            "cmd"
+          ]
+        },
+        "additionalProperties": false,
+        "required": [
+          "cmd"
+        ]
+      }
+    },
+    "additionalProperties": false,
+    "required": [
+      "name",
+      "description",
+      "run"
+    ]
+  },
+  "ENV": {
+    "type": "object",
+    "properties": {
+      "variables": {"type": "object"},
+      "labels": {"type": "object"},
+      "sources": {"type": "object"},
+      "dependencies": {
+        "type": "object",
+        "properties": {
+          "path": {
+            "type": "object",
+            "properties": {
+              "name": {"type": "string", "minLength": 1},
+              "path": {"type": "string", "minLength": 1}
+            },
+            "required": [
+              "name",
+              "path"
+            ],
+            "additionalProperties": false
+          },
+          "git": {
+            "type": "object",
+            "properties": {
+              "name": {"type": "string", "minLength": 1},
+              "path": {"type": "string", "minLength": 1},
+              "url": {"type": "string", "minLength": 1}
+            },
+            "required": [
+              "name",
+              "path",
+              "url"
+            ],
+            "additionalProperties": false
+          },
+          "spack": {
+            "type": "object",
+            "properties": {
+              "name": {"type": "string", "minLength": 1},
+              "package_name": {"type": "string", "minLength": 1}
+            },
+            "required": [
+              "type",
+              "package_name"
+            ],
+            "additionalProperties": false
+          }
+        }
+      }
+    },
+    "additionalProperties": false
+  }
```

Comment:
I was testing the various sections of the spec and noticed that the `paths` and `git` dependencies weren't being caught. When debugging it I found that the two sections were not specified as `array` types. The suggestion below caught the correct types and all required fields.

```suggestion
  "ENV": {
    "type": "object",
    "properties": {
      "variables": {"type": "object"},
      "labels": {"type": "object"},
      "sources": {"type": "object"},
      "dependencies": {
        "type": "object",
        "properties": {
          "paths": {
            "type": "array",
            "items": {
              "type": "object",
              "properties": {
                "name": {"type": "string", "minLength": 1},
                "path": {"type": "string", "minLength": 1}
              },
              "required": [
                "name",
                "path"
              ],
              "additionalProperties": false
            }
          },
          "git": {
            "type": "array",
            "items": {
              "properties": {
                "name": {"type": "string", "minLength": 1},
                "path": {"type": "string", "minLength": 1},
                "url": {"type": "string", "minLength": 1}
              },
              "required": [
                "name",
                "path",
                "url"
              ],
              "additionalProperties": false
            }
          },
          "spack": {
            "type": "object",
            "properties": {
              "name": {"type": "string", "minLength": 1},
              "package_name": {"type": "string", "minLength": 1}
            },
            "required": [
              "type",
              "package_name"
            ],
            "additionalProperties": false
          }
        }
      }
    },
    "additionalProperties": false
  }
```